### PR TITLE
Merge decentralized cpud to cpud

### PR DESCRIPTION
**Pull request by rminnich**
untested (yet)

### CODE REVIEW

**Comment by rminnich** [comment]
although, our tests pass, which should mean it all works fine ...

**Comment by rminnich** [comment]
OK I've confirmed that this works for "cpu classic" -- no mDNS -- when it is built with and without -tags mDNS
so far it's "no worse than before"

**Comment by rminnich** [comment]
This also appears to be working for decpud uses now.

**Review by brho — APPROVED** [review]

File: cmds/cpud/main_mdns_linux.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	dsInterface = flag.String("dsInterface", "", "DNSSD Interface")
+	dsTxtStr    = flag.String("dsTxt", "", "DNSSD key-value pair string parameterizing advertisement")
+	dsTxt       map[string]string
+)
+
+func runDS(debug bool) {
+	if debug {
+		ds.Verbose(log.Printf)
+	}
+		dsTxt = ds.ParseKv(*dsTxtStr)
```
Comment (brho) [inline cmds/cpud/main_mdns_linux.go]: extra tab?

File: cmds/cpud/main_mdns_linux.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+func serve(cpud string) error {
+	ln, err := listen(*network, *port)
+	if err != nil {
+		return err
+	}
+
+
+		v("Advertising w/dnssd %q", dsTxt)
```
Comment (brho) [inline cmds/cpud/main_mdns_linux.go]: whole block of extra tabs?

Comment (rminnich) [reply cmds/cpud/main_mdns_linux.go]: weird. that's from gofmt. I'll take a look.

File: cmds/cpud/main_mdns_linux.go (hunk 1 above)
Comment (rminnich) [reply cmds/cpud/main_mdns_linux.go]: ah, fixed by gofmt I guess.

**Comment by rminnich** [comment]
This is working fine as a cpu server. I went with a squash commit as the squash was mainly deletions with very little new code; further, some of the intermediate commits won't build correctly.

**Pull request merged**